git_machete/cmd.py

```diff
@@ -648,31 +649,131 @@ def annotate(b: str, words: List[str]) -> None:
+        remote: parse_github_remote_url(url) for remote, url in url_for_remote.items() if is_github_remote_url(url)
+    }
+    if not org_and_repo_for_github_remote:
         raise MacheteException(fmt('Remotes are defined for this repository, but none of them corresponds to GitHub (see `git remote -v` for details)'))
 
+    if len(org_and_repo_for_github_remote) == 1:
+        return list(org_and_repo_for_github_remote.items())[0]
+
+    if 'origin' in org_and_repo_for_github_remote:
+        return 'origin', org_and_repo_for_github_remote['origin']
+
+    raise MacheteException(f'Multiple non-origin remotes correspond to GitHub in this repository: '
+                           f'{", ".join(org_and_repo_for_github_remote.keys())}, aborting')
+
+
+def create_github_pr(cli_ctxt: CommandLineContext, head: str, draft: bool) -> None:
+    global annotations, up_branch
+    from git_machete.github import GitHubPullRequest, add_assignees_to_pull_request, add_reviewers_to_pull_request, \
+        create_pull_request, derive_current_user_login, set_milestone_of_pull_request
+
+    base: Optional[str] = up_branch.get(head)
+    if not base:
+        raise MacheteException(f'Branch {head} does not have a parent branch (it is a root), base branch for the PR cannot be established')
+
     org: str
     repo: str
-    if len(org_name_for_github_remote) == 1:
-        org, repo = list(org_name_for_github_remote.values())[0]
-    elif len(org_name_for_github_remote) > 1:
-        if 'origin' in org_name_for_github_remote:
-            org, repo = org_name_for_github_remote['origin']
-        else:
-            raise MacheteException(f'Multiple non-origin remotes correspond to GitHub in this repository: '
-                                   f'{", ".join(org_name_for_github_remote.keys())}, aborting')
+    _, (org, repo) = derive_remote_and_github_org_and_repo(cli_ctxt)
     current_user: Optional[str] = derive_current_user_login()
-    debug(cli_ctxt, 'sync_annotations_to_github_prs()', 'Current GitHub user is ' + (current_user or '<none>'))
+    debug(cli_ctxt, f'create_github_pr({head})', f'organization is {org}, repository is {repo}')
+    debug(cli_ctxt, f'create_github_pr({head})', 'current GitHub user is ' + (current_user or '<none>'))
+
+    def slurp_file_or_empty(path: str) -> str:
+        try:
+            return open(path).read()
+        except IOError:
+            return ''
+
+    title: str = popen_git(cli_ctxt, "log", "-1", "--format=%s").strip()
+    description_path = get_git_subpath(cli_ctxt, 'info', 'description')
+    description: str = slurp_file_or_empty(description_path)
+
+    ok_str = ' <green><b>OK</b></green>'
+    print(fmt(f'Creating a {"draft " if draft else ""}PR from `{head}` to `{base}`...'), end='', flush=True)
+    pr: GitHubPullRequest = create_pull_request(org, repo, head=head, base=base, title=title, description=description, draft=draft)
+    print(fmt(f'{ok_str}, see <b>{pr.html_url}</b>'))
```

Comment:
Also:

```
Creating a PR from chore/tidy-readme-comparison-table to main...
GitHub API returned 307 HTTP status with error message: Moved Permanently
Make sure that the token provided in gh auth status or ~/.config/hub or GITHUB_TOKEN is valid and allows for access to POST https://api.github.com/repos/VirtusLab/akka-safer-serializer/pulls`.
```

for a Github repo that has been renamed... we should follow the redirect in such case rather than fail the entire flow.

Reply:
Current library http.client used for requests seems to support redirections only if it gets status code 302 in response.
Let's consider using urllib instead in this case, which has built in support of redirections for codes 301, 302, 303, or 307.

Reply:
Whoops... I knew that `http.client` is a library of _plywood and cardboard_ (aka _z dykty i kartonu_), but didn't know it's that bad 😅 yeah, if you could migrate the existing impl to urllib then... as long as we don't add any external dependencies I'm fine.